gpytorch/kernels/periodic_kernel.py

```diff
@@ -122,10 +124,22 @@ def _set_period_length(self, value):
         self.initialize(raw_period_length=self.raw_period_length_constraint.inverse_transform(value))
 
     def forward(self, x1, x2, diag=False, **params):
-        x1_ = x1.div(self.period_length).mul(math.pi)
-        x2_ = x2.div(self.period_length).mul(math.pi)
-        diff = x1_.unsqueeze(-2) - x2_.unsqueeze(-3)
-        res = diff.sin().pow(2).sum(dim=-1).div(self.lengthscale).mul(-2.0).exp_()
+        # Pop this argument so that we can manually sum over dimensions
+        last_dim_is_batch = params.pop("last_dim_is_batch", False)
+        # Get lengthscale
+        lengthscale = self.lengthscale
+
+        x1_ = x1.div(self.period_length / math.pi)
+        x2_ = x2.div(self.period_length / math.pi)
+        diff = self.covar_dist(x1_, x2_, diag=diag, last_dim_is_batch=True, **params)
```

Comment:
Should this be the following?
```suggestion
        diff = self.covar_dist(x1_, x2_, diag=diag, last_dim_is_batch=last_dim_is_batch, **params)
```

Reply:
This should be `True`, since we end up manually summing over the dimensions of the kernel. I added an explanatory comment.